### ISSUE TITLE
Unable to display org file without any headers

### ISSUE THREAD

**Issue by joelmccracken** [200ok-ch/organice#583]
I have a file for which I receive a parse error:

```
Notes for self


ef.el contains code
.dir locals automaticaly loads this file


defun ef/print-files-with-new-versions
defun ef/print-all-files
```

Basically this is just a plain text file with the extension `.org`. I would expect it to be able to render this as a plain text file. Often my org mode files start out as a dump of text and I knock it into a structure when I have time.

**Comment by joelmccracken** [comment]
Problem appears to be here

https://github.com/200ok-ch/organice/blob/master/src/components/OrgFile/index.js#L458

it displays an error message if there are no headers, not necessarily if there is a parse error. So actually parser is seemingly fine.

**Comment by munen** [comment]
Hi @joelmccracken 

Thank you for checking out organice. What you describe is a known restriction in organice and documented in the README: 
https://github.com/200ok-ch/organice#current-restrictionsexpectations-of-organice

All actions in organice focus around headers, so it doesn't make sense to display a file without headers, because you couldn't do anything with it. You can check out the UX paradigm of organice with the sample file here: https://organice.200ok.ch/sample

Are you interested in proposing how to handle a file such as yours?

**Comment by munen** [comment]
The only option that I can think of is force the content of a file without a header into an implicit header. That would probably work. 

On export, it would add a `*` to the beginning of your file. However, that would violate a principle of organice which is not to create a diff on unchanged content of a viewed/edited Org file. 

The exporter could then in turn be changed to handle this edge case, too, and remove the star with empty title and just dump the rawDescription. 

@joelmccracken Would you be interested in implementing this?

**Comment by joelmccracken** [comment]
Yeah I can implement it. I'm trying to give org-mode another go as my primary means of self-organization, and I'm thinking that organice will be a big part of it. I was planning on hosting my own, so its either I fork and fix it for myself, or I just stick a header at the top of every org file I have in my repo. I'll poke around and see if I can figure it out.

**Comment by munen** [comment]
I meant not to ask if you want to fork and fix it for yourself, but to fix it upstream. 

There’s also no need to fork for self hosting. Here’s documentation for deploying yourself, there’s plenty of options: https://organice.200ok.ch/documentation.html#deployment

**Comment by munen** [comment]
In any case, good luck with getting organized with Org mode!(;

**Comment by joelmccracken** [comment]
Oh, that's actually what I meant. I'm glad if you'll accept it upstream; I was just saying that I was planning on fixing it for myself somehow anyway.

**Comment by munen** [comment]
Perfect! Let me know if I can be of help^^

**Comment by jzohrab** [comment]
Hi @munen , I opened a draft PR for this, https://github.com/200ok-ch/organice/pull/708, with some notes.  Happy to discuss.

**Comment by munen** [comment]
Hi @jzohrab 

Thank you for taking up this issue and apologies for the late response :pray: 

I'll respond directly in the PR, because you've added some additional context and questions in the PR.

**Comment by munen** [comment]
Responding to part of the PR commentary here, because it's part of the bigger UX discussion: >  I'm actually not sure how the code _should_ handle the case where there aren't any headers. Since a `.org` file is sort-of-assumed to have a header, maybe the existing failure is actually legit, because adding a dummy line like this code does, or changing structure, could somehow influence an existing org file in a bad way.

Generally speaking, Org mode files absolutely are valid without any headers. An empty file is a legit Org file, a file just including a single word or `- test` is a legit Org file.

Having said so, all actions in organice focus around headers, so it doesn't make sense to display a file without headers, because you couldn't do anything with it. That's why we have a safeguard in there.

The best option to solve this issue would be a complete UI overhaul to overcome this issue. There is an existing PR for "Experimental ui overhaul without inline editing": https://github.com/200ok-ch/organice/pull/559

Adding a 'dummy' header that will be ignored on export is a workaround. A quite ugly one at that, because the shown file isn't the file saved. Hence, the user should be properly aware of what's happening. Crashing on a file without headers is also very ugly, of course.

To make sure the user understands what's happening, I'm proposing this workflow:

![organice-ux-proposal-file-without-headers](https://user-images.githubusercontent.com/505721/125192846-156f3900-e24a-11eb-9174-8cb1dfbb43fe.png)

Additionally, we would document in the 'sample' what "Add a header to interact with this file." means. The meaning is: Add a new first header and group all existing content under this header. This is a regular header and it will also be included in the export.

What's your stance on this? To me, it's the best solution after months of background thinking apart from the complete UI overhaul linked to above.

NB: The mockup has been created with excalidraw: If you want to make adjustments, check it out here, export a PNG and save as a new URL: https://excalidraw.com/#json=5168599819878400,h-PX4uy4I3Uy96fLrHWd9w

**Comment by munen** [comment]
I've updated the documentation to include how to work with (excalidraw) mockups to make talking about UX easier: https://organice.200ok.ch/documentation.html#mockups

**Comment by munen** [comment]
@jzohrab @joelmccracken @schoettl Do you think the [proposal above](https://github.com/200ok-ch/organice/issues/583#issuecomment-877782251) is viable for your use cases?

**Comment by munen** [comment]
Slight UX update:

![image](https://user-images.githubusercontent.com/505721/125194157-65e99500-e250-11eb-810c-678e1f4db982.png)

**Comment by munen** [comment]
UX Update for clarity:

![image](https://user-images.githubusercontent.com/505721/125194376-5585ea00-e251-11eb-8f65-54f93e17a295.png)

**Comment by schoettl** [comment]
Is the checkmark icon in the plus button intended? Normally it's an x when this floating button menu is opened. I can't interpret the meaning of the checkmark.

Beside that, I think this would be a good, pragmatic way to solve the problem.

**Comment by munen** [comment]
> Is the checkmark icon in the plus button intended? Normally it's an x when this floating button menu is opened. I can't interpret the meaning of the checkmark.

*hihi*, adding mockups to make the meaning clearer or less clear, that is the question(;

Its unintended^^

> Beside that, I think this would be a good, pragmatic way to solve the problem.

Nice, thanks! 👍

**Comment by jzohrab** [comment]
I can't grok the difference between the 3rd screen and the 4th.  Urgent message shown => user clicks "+" => ... would it not go directly to the 4th screen?

I haven't thought deeply about the implications for the underlying data structures, but this seems like a low-friction way to get organice to work orga-nicely (sorry) in this almost-an-edge-case.

**Comment by munen** [comment]
@jzohrab 

> I can't grok the difference between the 3rd screen and the 4th. Urgent message shown => user clicks "+" => ... would it not go directly to the 4th screen?

No, it wouldn't. The '+' usually contains the capture templates. Clicking it will expand them. For example, in one of my files, it'll look like this:

![image](https://user-images.githubusercontent.com/505721/125333444-8ac23300-e34a-11eb-9357-94cc934031a8.png)

> I haven't thought deeply about the implications for the underlying data structures, but this seems like a low-friction way to get organice to work orga-nicely (sorry) in this almost-an-edge-case.

Nice. It's agreed, then.

@jzohrab Are you interested in picking this issue up?

**Comment by munen** [comment]
Closed by #853.
